lessons/beginners/list/index.md

```diff
@@ -30,204 +20,253 @@ print(cisla)
-seznam můžeme procházet po jednotlivých prvcích:
+seznam obsahuje sekvenci jakýchkoli hodnot.
+V našem případě obsahuje sekvenci řetězců.
+
+A tak jako můžeš pomocí cyklu `for` procházet řetězec po znacích,
+seznam můžeš procházet po jednotlivých prvcích:
 
 ```python
-for cislo in cisla:
-    print(cislo)
+for zviratko in zviratka:
+    print(zviratko)
 ```
 
-Seznamy se v programech vyskytují velice často:
-soubor se dá načíst jako seznam řetězců
-s jednotlivými řádky,
-seznam řetězců jako `'7♥'`
-a `'K♣'` může posloužit jako balíček karet,
-matematika je plná číselných řad,
-každá online služba má seznam uživatelů.
+Seznamy se v programech vyskytují často:
+soubor se dá načíst jako seznam řetězců s jednotlivými řádky,
+seznam řetězců jako `'7♥'` a `'K♣'` může posloužit jako balíček karet,
+matematika je plná číselných řad, každá online služba má seznam uživatelů.
+
+Hodnoty v seznamu můžou být jakéhokoli typu:
 
+```python
+prvni_prvocisla = [2, 3, 5, 7, 11]
+```
 
-Hodnoty v seznamu můžou být jakéhokoli typu,
-dokonce můžeme různé typy míchat v jednom seznamu
-(i když s takovými namixovanými seznamy se
-příliš často nesetkáme – více se používají v
-<var>n</var>-ticích, o kterých si povíme později):
+Dokonce můžeš různé typy míchat v jednom seznamu.
+(S takovými namixovanými seznamy se ovšem příliš často nesetkáš.
+Různé typy hodnot se používají spíš v <var>n</var>-ticích, o kterých si povíme
+později):
 
 ```python
 seznam = [1, 'abc', True, None, range(10), len]
 print(seznam)
 ```
 
-## Vybírání ze seznamů
+## Neměnitelné hodnoty
+
+Důležitá vlastnost seznamů je, že se dají *měnit*.
+
+Než vysvětlíme o co jde, připomeňme si jak fungují hodnoty, které se měnit
+nedají – např. čísla, řetězce, `True`/`False`/`None`.
 
-Nejzákladnější operaci se seznamy,
-cyklus `for`, už jsme si ukázal{{gnd('i', 'y', both='i')}}.
-Druhá nejdůležitější operace je vybírání
-jednotlivých prvků.
-To funguje jako u řetězců: do hranatých závorek
-se dá číslo prvku. Čísluje se, jako u řetězců,
-od nuly; záporná čísla označují prvky od konce.
+Vyzkoušej si následující kousek kódu. Co je na něm „špatně“?
 
 ```python
-print(cisla[2])
+kamaradka = 'Žaneta'
+print(kamaradka)
+kamaradka.upper()
+print(kamaradka)
 ```
 
-Hranatými závorkami můžeme získávat podseznamy.
-[Diagram z materiálů k řetězcům]({{ lesson_url('beginners/str')}}#slicing-diagram)
-ukazuje, jak u takového „sekání” číslovat:
-funguje to stejně, jen místo menšího řetězce
-dostaneme menší seznam.
+Proměnná `kamaradka` obsahuje řetězec `'Žaneta'` (který se už nedá změnit).
+Metoda `upper()` vytvoří a vrátí *nový* řetězec `'ŽANETA'`.
+Výsledná hodnota se ale v našem programu nevyužije – Python ji vypočítá,
+ale pak na ni „zapomene“.
+
+Oprava je snadná: výsledek si ulož do proměnné.
+Často budeš chtít takový výsledek uložit zpátky do původní proměnné:
 
 ```python
-print(cisla[2:-3])
+kamaradka = kamaradka.upper()
 ```
 
-## Měnění seznamů
+Tímto přiřazením Python „zapomene“ na původní hodnotu,
```

Comment:
Just a tiny detail, the word `„zapomene“` is here twice, and they are somewhat close to each other. But I don't think that many people will see it as a problem. :smile: The word `„zahodí“` might work as an alternative for one of the two occurrences.

Reply:
Actually, I'd rather use the same word here to emphasize that the same thing is happening. Perhaps „zahodí“ is better for *both* occurrences, though.